File: src/main/java/com/pawan/choure/java/handler/TradeMessageFilter.java
package com.pawan.choure.java.handler;

import org.springframework.integration.MessageRejectedException;
import org.springframework.integration.core.MessageSelector;
import org.springframework.messaging.Message;

public class TradeMessageFilter implements MessageSelector {
    @Override
    public boolean accept(Message<?> message) {
        Object payload =message.getPayload();
        if(payload instanceof String){
            return true;
        }else {
            throw  new MessageRejectedException(message,"Unknown Data Type has been received");
        }
    }
}

File: src/main/kotlin/com/pawan/choure/kotlin/handler/TradeMessageFilter.kt
package com.pawan.choure.kotlin.handler

import org.springframework.integration.MessageRejectedException
import org.springframework.integration.core.MessageSelector
import org.springframework.messaging.Message

class TradeMessageFilter:MessageSelector {
    open override fun accept(message: Message<*>?): Boolean {
        val payload = message!!.payload
        return if (payload is String) {
            true
        } else {
            throw MessageRejectedException(message, "Unknown Data Type has been received")
        }
    }
}